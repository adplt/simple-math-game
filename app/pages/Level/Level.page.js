import React, {Component} from 'react';
import {
  Text,
  View,
} from 'react-native';
import styles from './Level.page.style';
import Touchable from '../../components/Touchable/Touchable.component';
import PropTypes from 'prop-types';

export default class Level extends Component {

  static propTypes = {
    navigation: PropTypes.object,
    goBack: PropTypes.func
  }

  onPress = (params) => {
    const {navigation} = this.props;
    navigation.navigate('Game', {...navigation.state.params, ...params});
  }

  plus = () => this.onPress({operator: '+'});

  minus = () => this.onPress({operator: '-'});

  multiple = () => this.onPress({operator: '*'});

  divide = () => this.onPress({operator: '/'});

  render () {
    const {goBack} = this.props;
    return (
      <View style={styles.container}>
        <Text style={styles.welcome}>
          {'Choose Option'}
        </Text>
        <Touchable onPress={this.plus} style={styles.menuTouchable}>
          <Text style={styles.instructions}>{'Plus'}</Text>
        </Touchable>
        <Touchable onPress={this.minus} style={styles.menuTouchable}>
          <Text style={styles.instructions}>{'Minus'}</Text>
        </Touchable>
        <Touchable onPress={this.multiple} style={styles.menuTouchable}>
          <Text style={styles.instructions}>{'Multiple'}</Text>
        </Touchable>
        <Touchable onPress={this.divide} style={styles.menuTouchable}>
          <Text style={styles.instructions}>{'Divide'}</Text>
        </Touchable>
        <Touchable onPress={goBack} style={styles.menuTouchable}>
          <Text style={styles.instructions}>{'Back'}</Text>
        </Touchable>
      </View>
    );
  }
}
