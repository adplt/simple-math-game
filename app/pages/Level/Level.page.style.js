export default {
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#FFF',
    margin: 40,
    opacity: 0.8
  },
  welcome: {
    fontSize: 40,
    textAlign: 'center',
    margin: 10,
    color: 'black'
  },
  instructions: {
    textAlign: 'left',
    alignItems: 'flex-start',
    color: '#333333',
    marginBottom: 5
  },
  row: {
    flexDirection: 'row'
  },
  menuTouchable: {
    padding: 10
  },
  background: {
    flex: 1
  }
};
