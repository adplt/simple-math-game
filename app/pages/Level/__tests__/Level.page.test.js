import React from 'react';
import renderer from 'react-test-renderer';
import Level from '../Level.page';

describe('Level component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Level />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
