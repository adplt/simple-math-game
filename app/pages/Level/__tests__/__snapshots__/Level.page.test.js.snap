// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Level component renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFF",
        "borderRadius": 30,
        "flex": 1,
        "justifyContent": "center",
        "margin": 40,
        "opacity": 0.75,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "black",
          "fontSize": 40,
          "margin": 10,
          "textAlign": "center",
        }
      }
    >
      Choose Option
    </Text>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "padding": 10,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "alignItems": "flex-start",
            "color": "#333333",
            "marginBottom": 5,
            "textAlign": "left",
          }
        }
      >
        Plus
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "padding": 10,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "alignItems": "flex-start",
            "color": "#333333",
            "marginBottom": 5,
            "textAlign": "left",
          }
        }
      >
        Minus
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "padding": 10,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "alignItems": "flex-start",
            "color": "#333333",
            "marginBottom": 5,
            "textAlign": "left",
          }
        }
      >
        Multiple
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "padding": 10,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "alignItems": "flex-start",
            "color": "#333333",
            "marginBottom": 5,
            "textAlign": "left",
          }
        }
      >
        Divide
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "padding": 10,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "alignItems": "flex-start",
            "color": "#333333",
            "marginBottom": 5,
            "textAlign": "left",
          }
        }
      >
        Back
      </Text>
    </View>
  </View>
</View>
`;
